### ISSUE TITLE
Guid() mixup with 128bit UUIDs

### ISSUE THREAD

**Issue by kschulz-samsung** [pauldemarco/flutter_blue#390]
On android I get the full 128bit service uuid in the scan result, but on IOS I only get the differing part of the UUID (16bits). This means that when I want to do a comparison, I would normally do

Guid(result.serviceUUID) ==  Guid(myUUID) 
This works perfectly on Android but on  iOS it will crash as Guid will not take a string of length 4 bytes as input (the string representation of the 16bit uuid). 

It would be nice if Guid() could handle both formats as it is basically just inserting the short uuid into the generic 128bit UUID placeholder.

**Comment by pauldemarco** [comment]
I agree.  The GUID should accept the shorter version of the uuid and auto-fill with the Bluetooth sig data, so that we can get equality checks working.  I’ll look into this.
